javaagent-api/src/main/jflex/SqlSanitizer.flex

```diff
@@ -0,0 +1,378 @@
+        // YYEOF token may be used to stop processing
+        if (token == YYEOF) {
+          break;
+        }
+      }
+      return sanitizer.getResult();
+    } catch (java.io.IOException e) {
+      // should never happen
+      return new SqlStatementInfo(null, null, null);
+    }
+  }
+
+  // max length of the sanitized statement - SQLs longer than this will be trimmed
+  private static final int LIMIT = 32 * 1024;
+
+  private final StringBuilder builder = new StringBuilder();
+
+  private void appendCurrentFragment() {
+    builder.append(zzBuffer, zzStartRead, zzMarkedPos - zzStartRead);
+  }
+
+  private boolean isOverLimit() {
+    return builder.length() > LIMIT;
+  }
+
+  // you can reference a table in the FROM clause in one of the following ways:
+  //   table
+  //   table t
+  //   table as t
+  // in other words, you need max 3 identifiers to reference a table
+  private static final int FROM_TABLE_REF_MAX_IDENTIFIERS = 3;
+
+  private int parenLevel = 0;
+  private boolean insideComment = false;
+  private Operation operation = NoOp.INSTANCE;
+  private boolean extractionDone = false;
+
+  private void setOperation(Operation operation) {
+    if (this.operation == NoOp.INSTANCE) {
+      this.operation = operation;
+    }
+  }
+
+  private static abstract class Operation {
+    String mainTable = null;
+
+    /** @return true if all statement info is gathered */
+    boolean handleFrom() {
+      return false;
+    }
+
+    /** @return true if all statement info is gathered */
+    boolean handleInto() {
+      return false;
+    }
+
+    /** @return true if all statement info is gathered */
+    boolean handleJoin() {
+      return false;
+    }
+
+    /** @return true if all statement info is gathered */
+    boolean handleIdentifier() {
+      return false;
+    }
+
+    /** @return true if all statement info is gathered */
+    boolean handleComma() {
+      return false;
+    }
+
+    SqlStatementInfo getResult(String fullStatement) {
+      return new SqlStatementInfo(fullStatement, getClass().getSimpleName().toUpperCase(), mainTable);
+    }
+  }
+
+  private static class NoOp extends Operation {
+    static final Operation INSTANCE = new NoOp();
+
+    SqlStatementInfo getResult(String fullStatement) {
+      return new SqlStatementInfo(fullStatement, null, null);
+    }
+  }
+
+  private class Select extends Operation {
+    // you can reference a table in the FROM clause in one of the following ways:
+    //   table
+    //   table t
+    //   table as t
+    // in other words, you need max 3 identifiers to reference a table
+    private static final int FROM_TABLE_REF_MAX_IDENTIFIERS = 3;
+
+    boolean expectingTableName = false;
+    boolean mainTableSetAlready = false;
+    int identifiersAfterMainFromClause = 0;
+
+    boolean handleFrom() {
+      if (parenLevel == 0) {
+        // main query FROM clause
+        expectingTableName = true;
+        return false;
+      } else {
```

Comment:
would prefer to remove redundant `else` when `if{}` returns.

Reply:
I think this may be subjective, and I don't think there's an established convention in this codebase (I've seen lots of both styles). Should we add to https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/docs/contributing/style-guideline.md? IIRC @jkwatson has expressed this preference previously as well.